sdk/core/System.Net.ClientModel/src/Message/MessageBody.cs

```diff
@@ -173,63 +164,100 @@ private async Task<Stream> ToStreamSyncOrAsync(CancellationToken cancellationTok
         /// <inheritdoc/>
         public abstract void Dispose();
 
-        // TODO: Note, this is copied from RequestContent.  When we can remove the corresponding
-        // shared source file, we should make sure there is only one copy of this moving forward.
-        private sealed class JsonModelMessageBody : MessageBody
+        private sealed class ModelMessageBody : MessageBody
```

Comment:
@m-nash, I'd love your eyes on this change addressing Stephen Toub's feedback that the MessageBody.Create(IJsonModel) overload ties the protocol-agnostic MesageBody type to a protocol (JSON).  This should be functionally equivalent to what was here before, but please shout if you spot errors in the implementation!